vendor-extra/ContentPageBundle/src/Controller/ContentController.php

```diff
@@ -49,25 +51,52 @@ function (ResponseInterface $response) use ($request) : Response {
                     $dom = FluentDOM::load((string) $response->getBody());
                     $dom->registerNamespace('libero', 'http://libero.pub');
 
+                    /** @var DOMNodeList|Element[] $frontList */
+                    $frontList = $dom('/libero:item/libero:front[1]');
+                    $front = $frontList->item(0);
+
+                    if (!$front instanceof Element) {
+                        throw new UnexpectedValueException('Could not find a front');
+                    }
+
                     /** @var string $title */
-                    $title = $dom('string(/libero:item/libero:front/libero:title)');
+                    $title = $front('string(libero:title)');
 
                     if ('' === $title) {
                         throw new UnexpectedValueException('Could not find a title');
                     }
 
                     $context = [
                         'lang' => $request->getLocale(),
-                        'dir' => 'right-to-left' === Misc::getCharacterOrder($request->getLocale()) ? 'rtl' : 'ltr',
+                        'dir' => $this->getDirection($request->getLocale()),
                     ];
 
+                    $contentHeader = [
+                        'template' => '@LiberoPatterns/content-header.html.twig',
```

Comment:
yes, abstracting all the templates that need to be used for each block will be an interesting design to introduce here